src/qca/utils/utils.py

```diff
@@ -358,23 +359,30 @@ def grab_circuit_resources(circuit: cirq.AbstractCircuit,
                            algo_name: str,
                            fname: str,
                            is_extrapolated:bool,
+                           use_analytical: bool = False,
                            nsteps: int|None=None,
                            bits_precision:int|None=None,
                            metadata: EstimateMetaData|None=None,
                            write_circuits:bool=False,
                            include_nested_resources:bool=False,
                            gate_synth_accuracy: int | float=10) -> None:
-    estimates = circuit_estimate(
-        circuit=circuit,
-        outdir=outdir,
-        nsteps=nsteps,
-        algo_name=algo_name,
-        is_extrapolated=is_extrapolated,
-        include_nested_resources=include_nested_resources,
-        gate_synth_accuracy=gate_synth_accuracy,
-        bits_precision=bits_precision,
-        write_circuits=write_circuits,
-    )
+    if not use_analytical:
+        estimates = circuit_estimate(
+            circuit=circuit,
+            outdir=outdir,
+            nsteps=nsteps,
+            algo_name=algo_name,
+            is_extrapolated=is_extrapolated,
+            include_nested_resources=include_nested_resources,
+            gate_synth_accuracy=gate_synth_accuracy,
+            bits_precision=bits_precision,
+            write_circuits=write_circuits,
+        )
+    else:
+        estimates = estimate_resources(
```

Comment:
Question, will we ever need circuit_precision in our resourse estimates? I just recalled that PyLIQTRs estimate_resources also includes an argument for circuit_precision as well.